promscale/send-data/remote-write.md

```diff
@@ -0,0 +1,302 @@
+the label value.
+
+The `samples` array must be in the following format:
+
+*   Each array can only contain two values.
+*   The first value is a UNIX timestamp in milliseconds. It must be an integer.
+*   The second value is a floating point number which represents the metric
+    value for that timestamp.
+
+To send this data to Promscale, send an HTTP POST request. Set the request body
+to the JSON payload. Also set these HTTP headers:
+
+*   `Content-Type`: `application/json`.
+*   `Content-Encoding`: Set to `snappy` if using snappy compression. Otherwise,
+    leave unset.
+
+Here's an example request sent with `curl`:
+
+```bash
+curl --header "Content-Type: application/json" \
+--request POST \
+--data '{"labels":{"__name__":"foo"},"samples":[[1577836800000, 100]]}' \
+"http://localhost:9201/write"
+```
+
+Here's an example request using snappy encoding:
+
+```bash
+curl --header "Content-Type: application/json" \
+--header "Content-Encoding: snappy" \
+--request POST \
+--data-binary "@snappy-payload.sz" \
+"http://localhost:9201/write"
+```
+
+## Prometheus and OpenMetric text format
+
+The text format makes it easier to ingest samples using a push model. Metrics
+exposed in this format can be directly forwarded to Promscale. Promscale parses
+the text file and stores the parsed data.
+
+For more details about the format, see the:
+
+*   [Prometheus text-based format docs][prometheus-text-format]
+*   [OpenMetrics format specification][openmetrics-format]
+
+A simple payload might look like this:
+
+```bash
+# HELP http_requests_total The total number of HTTP requests.
+# TYPE http_requests_total counter
+http_requests_total{method="post",code="200"} 1027 1395066363000
+http_requests_total{method="post",code="400"}    3 1395066363000
+# A histogram, which has a pretty complex representation in the text format:
+# HELP http_request_duration_seconds A histogram of the request duration.
+# TYPE http_request_duration_seconds histogram
+http_request_duration_seconds_bucket{le="0.05"} 24054
+http_request_duration_seconds_bucket{le="0.1"} 33444
+http_request_duration_seconds_bucket{le="0.2"} 100392
+http_request_duration_seconds_bucket{le="0.5"} 129389
+http_request_duration_seconds_bucket{le="1"} 133988
+http_request_duration_seconds_bucket{le="+Inf"} 144320
+http_request_duration_seconds_sum 53423
+http_request_duration_seconds_count 144320
+...
+```
+
+Each sample entry is a single line in this format:
+
+*   Metric name
+*   **OPTIONAL** Collection label name and value, in curly braces, immediately
+    following the metric name
+*   Floating point number that represents the actual measured value
+*   **OPTIONAL** An integer timestamp in milliseconds since the UNIX epoch. In
+    other words, the number of milliseconds since `1970-01-01 00:00:00 UTC`,
+    excluding leap seconds. This should be represented as required by Go's
+    [ParseInt][parseint] function.
+
+    If the timestamp is omitted, request time is used in its place.
```

Comment:
```suggestion
If the timestamp is omitted, request time is used in its place.
```

Reply:
Intentional indent to create a multi-paragraph list item (unless we don't like those? I don't think there's an existing rule about it in the style guide)

Reply:
Ah, my bad. Carry on 😅